src/main/java/Duke/Task/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package Duke.Task;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task{
+    protected String by;
+    protected LocalDate date;
+
+
+    /**
+     * Create a Deadline task with a date as deadline.
+     *
+     * @param description Description of task.
+     * @param by Date as deadline of task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        List<String> dateFormats = Arrays.asList("yyyy-MM-dd","dd-MM-yyyy", "dd/MM/yyyy", "dd:MM:yyyy");
+
+        DateTimeFormatter formatToParse;
+            for (int i = 0; i <= dateFormats.size(); i++) {
+                try {
+                    formatToParse = DateTimeFormatter.ofPattern(dateFormats.get(i));
+                    this.date = LocalDate.parse(by, formatToParse);
+                    break;
+                } catch (Exception wrongFormat) {
```

Comment:
This will also capture the Exception at line 31 when `i == dateFormats.size()`.
Perhaps, it might be a better practice to put the `if` statement at line 36 before `try...catch`, and catch only `DateTimeParseException`.

Reply:
Thanks for the suggestion. 
What I was intending to do here is that because the LocalDate.parse will always return exception if the format used is incorrect, I needed to catch the Exception to continue to try the next format, only when all formats are exhausted, I throw the exception. 
Hence its kind of difficult to put the statement outside(I have to admit this solution is kind of hard coded as well)😅